openmdao/core/implicitcomponent.py

```diff
@@ -71,6 +71,13 @@ def _solve_nonlinear(self):
             else:
                 return result
 
+    def _guess_nonlinear(self):
+        """
+        Provide initial guess for states.
+        """
+        with self._unscaled_context(outputs=[self._outputs], residuals=[self._residuals]):
+            self.guess_nonlinear(self._inputs, self._outputs, self._residuals)
```

Comment:
I do the check in Group, so that it never calls into here if the system doesn't provide a guess.